### DISCUSSION TITLE
tryWorkspaceByIdent vs tryWorkspaceByDescriptor inside plugin

### DISCUSSION THREAD

**Discussion by ikibalnyi** [yarnpkg/berry#2832] [Q&A]
I'm trying make a plugin and indside I need to get nested workspaces of the workspace and [`tryWorkspaceByDescriptor`](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/Project.ts#L456) stops working if I specify `peerDependencies` for nested workspace, while [`tryWorkspaceByIdent`](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/Project.ts#L438) seems like working.
Going deeper, after [devirtualization](https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/Project.ts#L458) the range is not accepted anymore (https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-core/sources/Project.ts#L462).
Could you please explain how peerDependencies affects the range?
And is that an expected behavior in general?

**Comment by merceyz** [comment]
`tryWorkspaceByDescriptor` only recently started devirtualizing the descriptor in https://github.com/yarnpkg/berry/pull/2675 which was released in `3.0.0-rc.1`, which version are you using?
